### PR TITLE
Code Modernization: Replace usage of strpos() with str_contains()

### LINKED ISSUE #32863: Code modernization: Replace usage of `strpos` with `str_contains`
Source: https://core.trac.wordpress.org/ticket/58206

Let's follow Core's lead once Jetpack requires WordPress 6.3.

Primary issue: #32865

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #32863

## Proposed changes:

`str_contains()` was introduced in PHP 8.0 to perform a case-sensitive check indicating if the string to search in (haystack) contains the given substring (needle).

WordPress core includes a polyfill for `str_contains()` on PHP < 8.0 as of WordPress 5.9.

This commit replaces `strpos()` with `str_contains()`, making the code more readable and consistent, as well as better aligned with modern development practices.

Related core ticket: https://core.trac.wordpress.org/ticket/58206

Note: this does not make changes in a few places:
- The Analyzer, Autoloader, Changelogger, WAF packages
- The VaultPress plugin since it can still be used on older versions of WordPress, where the `str_contains()` shim did not exist.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* Primary issue: #32865

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Activate all touched plugins and watch out for fatals on the frontend and wp-admin.

### CODE REVIEW

File: projects/packages/connection/tests/php/test_Manager_unit.php [hunk 1]
```diff
@@ -403,10 +403,10 @@ public function test_get_signed_token() {
...
 
 		$signed_token = ( new Tokens() )->get_signed_token( $access_token );
-		$this->assertTrue( strpos( $signed_token, 'token' ) !== false );
-		$this->assertTrue( strpos( $signed_token, 'timestamp' ) !== false );
-		$this->assertTrue( strpos( $signed_token, 'nonce' ) !== false );
-		$this->assertTrue( strpos( $signed_token, 'signature' ) !== false );
+		$this->assertTrue( str_contains( $signed_token, 'token' ) );
+		$this->assertTrue( str_contains( $signed_token, 'timestamp' ) );
+		$this->assertTrue( str_contains( $signed_token, 'nonce' ) );
+		$this->assertTrue( str_contains( $signed_token, 'signature' ) );
```
Comment (anomiex) [inline projects/packages/connection/tests/php/test_Manager_unit.php]: Probably these should just use `$this->assertStringContainsString()`.

Same elsewhere where `assertTrue( str_contains() )` or `assertFalse( str_contains() )` are being used.

File: projects/packages/image-cdn/src/class-image-cdn.php [hunk 2]
```diff
@@ -399,11 +399,11 @@ public static function filter_the_content( $content ) {
 					// First, check the image tag. Note we only check for pixel sizes now; HTML4 percentages have never been correctly
 					// supported, so we stopped pretending to support them in JP 9.1.0.
 					if ( preg_match( '#[\s"\']width=["\']?([\d%]+)["\']?#i', $images['img_tag'][ $index ], $width_string ) ) {
-						$width = false === strpos( $width_string[1], '%' ) ? $width_string[1] : false;
+						$width = ! str_contains( $width_string[1], '%' ) ? $width_string[1] : false;
```
Comment (anomiex) [inline projects/packages/image-cdn/src/class-image-cdn.php]: I wonder if we should flip these around to make it a little less confusing, like

```suggestion
						$width = str_contains( $width_string[1], '%' ) ? false : $width_string[1];
```

File: projects/packages/forms/src/contact-form/class-contact-form.php [hunk 3]
```diff
@@ -349,7 +349,7 @@ public static function parse( $attributes, $content ) {
 			 * @param int $id Contact Form ID.
 			 */
 			$url                     = apply_filters( 'grunion_contact_form_form_action', "{$url}#contact-form-{$id}", $GLOBALS['post'], $id );
-			$has_submit_button_block = ! ( false === strpos( $content, 'wp-block-jetpack-button' ) );
+			$has_submit_button_block = ! ( ! str_contains( $content, 'wp-block-jetpack-button' ) );
```
Comment (anomiex) [inline projects/packages/forms/src/contact-form/class-contact-form.php]:
```suggestion
			$has_submit_button_block = str_contains( $content, 'wp-block-jetpack-button' );
```

File: projects/packages/image-cdn/src/class-image-cdn.php [hunk 4]
```diff
@@ -399,11 +399,11 @@ public static function filter_the_content( $content ) {
...
 					}
 
 					if ( preg_match( '#[\s"\']height=["\']?([\d%]+)["\']?#i', $images['img_tag'][ $index ], $height_string ) ) {
-						$height = false === strpos( $height_string[1], '%' ) ? $height_string[1] : false;
+						$height = ! str_contains( $height_string[1], '%' ) ? $height_string[1] : false;
```
Comment (anomiex) [inline projects/packages/image-cdn/src/class-image-cdn.php]: (same, just leaving the suggestion here too)
```suggestion
						$height = str_contains( $height_string[1], '%' ) ? false : $height_string[1];
```

File: projects/packages/partner/changelog/update-strpos-str-contains
Comment (anomiex) [inline projects/packages/partner/changelog/update-strpos-str-contains:1]: This package has no changes other than this file?

File: projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php [hunk 5]
```diff
@@ -2912,7 +2912,7 @@ public static function parse( $attributes, $content ) {
 			 * @param int $id Contact Form ID.
 			 */
 			$url                     = apply_filters( 'grunion_contact_form_form_action', "{$url}#contact-form-{$id}", $GLOBALS['post'], $id );
-			$has_submit_button_block = ! ( false === strpos( $content, 'wp-block-jetpack-button' ) );
+			$has_submit_button_block = ! ( ! str_contains( $content, 'wp-block-jetpack-button' ) );
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php]:
```suggestion
			$has_submit_button_block = str_contains( $content, 'wp-block-jetpack-button' );
```

File: projects/plugins/jetpack/modules/shortcodes/recipe.php [hunk 6]
```diff
@@ -493,7 +493,7 @@ private static function output_list_content( $content, $type ) {
 		if (
 			strpos( $content, '&#8211;' ) !== false ||
 			strpos( $content, '&#8212;' ) !== false ||
-			strpos( $content, '-' ) !== false ||
+			str_contains( $content, '-' ) ||
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/shortcodes/recipe.php:496]: Why only this one and not all the rest in here?

File: projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php [hunk 7]
```diff
@@ -1197,7 +1197,7 @@ public static function migrate_post_type_rules() {
 $current_url = ! empty( $_SERVER['REQUEST_URI'] ) ? esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
 if ( is_customize_preview()
 	|| 'widgets.php' === $pagenow
-	|| ( false !== strpos( $current_url, '/wp-json/wp/v2/block-renderer' ) )
+	|| ( str_contains( $current_url, '/wp-json/wp/v2/block-renderer' ) )
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php]:
```suggestion
	|| str_contains( $current_url, '/wp-json/wp/v2/block-renderer' )
```

File: projects/plugins/jetpack/modules/widgets/facebook-likebox.php [hunk 8]
```diff
@@ -426,7 +426,7 @@ public function normalize_facebook_args( $args ) {
 	 * @param string $url URL to check.
 	 */
 	public function is_valid_facebook_url( $url ) {
-		return ( false !== strpos( $url, 'facebook.com' ) ) ? true : false;
+		return ( str_contains( $url, 'facebook.com' ) ) ? true : false;
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/widgets/facebook-likebox.php]:
```suggestion
		return str_contains( $url, 'facebook.com' );
```

File: projects/plugins/jetpack/modules/wordads/php/class-wordads-params.php [hunk 9]
```diff
@@ -119,7 +119,7 @@ public function __construct() {
 		}
 
 		$this->url = esc_url_raw( ( is_ssl() ? 'https' : 'http' ) . '://' . ( isset( $_SERVER['HTTP_HOST'] ) ? wp_unslash( $_SERVER['HTTP_HOST'] ) : 'localhost' ) . ( isset( $_SERVER['REQUEST_URI'] ) ? wp_unslash( $_SERVER['REQUEST_URI'] ) : '' ) );
-		if ( ! ( false === strpos( $this->url, '?' ) ) && ! isset( $_GET['p'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( ! ( ! str_contains( $this->url, '?' ) ) && ! isset( $_GET['p'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/wordads/php/class-wordads-params.php]:
```suggestion
		if ( str_contains( $this->url, '?' ) && ! isset( $_GET['p'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

File: projects/packages/connection/tests/php/test_Manager_unit.php (hunk 1 above)
Comment (jeherve) [reply projects/packages/connection/tests/php/test_Manager_unit.php]: Done in 336c6aef88e274c3d2c7cec0b297a8d6512a8bf0 and 30e98c67274a7aa142529c4fca6b90128bd4197f

File: projects/packages/partner/changelog/update-strpos-str-contains
Comment (jeherve) [reply projects/packages/partner/changelog/update-strpos-str-contains:1]: Indeed, not sure how that happened. Removed in 62ef6a10455cab3c7a3718b5af1224b7ab3912be

File: projects/plugins/jetpack/modules/shortcodes/recipe.php (hunk 6 above)
Comment (jeherve) [reply projects/plugins/jetpack/modules/shortcodes/recipe.php:496]: Done in 141bf1195491aef909482be92b19a7d0563a7985

File: projects/packages/partner/changelog/update-strpos-str-contains
Comment (jeherve) [reply projects/packages/partner/changelog/update-strpos-str-contains:1]: Indeed, not sure how that happened. Removed in 62ef6a10455cab3c7a3718b5af1224b7ab3912be

File: projects/plugins/jetpack/tests/php/modules/widgets/test_contact-info-widget.php [hunk 10]
```diff
@@ -32,7 +32,7 @@ public function test_form_apikey_field_with_no_filter() {
 		$this->contact_info_widget->form( null );
 		$output_string = ob_get_clean();
 
-		$apikey_field_displayed = false === strpos( $output_string, '<input type="hidden" id="widget-widget_contact_info' );
+		$apikey_field_displayed = ! str_contains( $output_string, '<input type="hidden" id="widget-widget_contact_info' );
 		$this->assertTrue( $apikey_field_displayed );
```
Comment (anomiex) [inline projects/plugins/jetpack/tests/php/modules/widgets/test_contact-info-widget.php]: Looks like the three in this file could be switched to `assertStringContainsString` or `assertStringNotContainsString` too.

Comment (jeherve) [reply projects/plugins/jetpack/tests/php/modules/widgets/test_contact-info-widget.php]: 👍 Done in 9f26940b3fa7a5f97869829ea041787461624558

**Review by anomiex — DISMISSED** [review]

File: projects/packages/search/tests/php/test-helpers.php
Comment (anomiex) [inline projects/packages/search/tests/php/test-helpers.php:1]: Looks like this test class will need to use the `Yoast\PHPUnitPolyfills\Polyfills\AssertStringContains` trait.

(the other files must be using a WordPress or WorDBless base class that brings in all the polyfills)

Comment (jeherve) [reply projects/packages/search/tests/php/test-helpers.php:1]: 👍 Done in 42ca014d7930cbf1af314312a1c47751ceba6510

File: projects/plugins/jetpack/tests/php/modules/widgets/test_contact-info-widget.php [hunk 11]
```diff
@@ -78,7 +76,7 @@ function ( $value ) {
 		$this->contact_info_widget->form( $instance );
 		$output_string = ob_get_clean();
 
-		$apikey_field_displayed = false === strpos( $output_string, '<input type="hidden" id="widget-widget_contact_info' );
+		$apikey_field_displayed = ! str_contains( $output_string, '<input type="hidden" id="widget-widget_contact_info' );
 		$this->assertTrue( $apikey_field_displayed );
```
Comment (anomiex) [inline projects/plugins/jetpack/tests/php/modules/widgets/test_contact-info-widget.php]: Missed one
```suggestion
		$this->assertStringNotContainsString( '<input type="hidden" id="widget-widget_contact_info', $output_string );
```

**Review by anomiex — APPROVED** [review]

**Pull request merged**